src/ArcherContainer.js

```diff
@@ -226,26 +248,40 @@ export class ArcherContainer extends React.Component<Props, State> {
     return [].concat.apply([], jaggedSourceToTargets);
   };
 
+  _createShapeObj = (style: LineType) => {
+    const chosenEndShape = getEndShapeFromStyle(style);
+
+    if (chosenEndShape === 'circle') {
+      return {
+        circle: {
+          ...this.props.endShape?.circle,
+          ...style.endShape?.circle,
+        },
+      };
+    } else {
+      return {
+        arrow: {
+          ...this.props.endShape?.arrow,
+          ...style.endShape?.arrow,
+        },
+      };
+    }
+  };
+
   _computeArrows = (): React$Element<typeof SvgArrow>[] => {
     const parentCoordinates = this._getParentCoordinates();
 
     return this._getSourceToTargets().map(
-      ({ source, target, label, style }: SourceToTargetType) => {
-        const strokeColor = (style && style.strokeColor) || this.props.strokeColor;
+      ({ source, target, label, style = {} }: SourceToTargetType) => {
+        const endShape = this._createShapeObj(style);
 
-        // Actual arrowLength value might be 0, which can't work with a simple 'actualValue || defaultValue'
-        let arrowLength = this.props.arrowLength;
-        if (style && (style.arrowLength || style.arrowLength === 0)) {
-          arrowLength = style.arrowLength;
-        }
+        const strokeColor = style.strokeColor || this.props.strokeColor;
```

Comment:
By assigning `style` to a default parameter of `{}`, we can remove all of the `style && style...` checks.